File: T/2073-time-needed-to-buy-tickets/main.go
package main

import (
	"fmt"
	"sort"
)

func timeRequiredToBuy(tickets []int, k int) int {
	prev := 0
	base := tickets[k]
	for i := 0; i < k; i++ {
		if tickets[i] >= tickets[k] {
			prev++
		}
	}
	sort.Ints(tickets)

	idx := 0
	seconds := 0
	for i := 0; i < len(tickets); i++ {
		if tickets[i] < base {
			seconds += tickets[i]
			continue
		}
		idx = i
		break
	}

	seconds += (len(tickets)-idx)*(base-1) + prev + 1
	return seconds
}

func main() {
	fmt.Println()
}

File: T/2073-time-needed-to-buy-tickets/main.rs
struct Solution {}

impl Solution {
  pub fn time_required_to_buy(tickets: Vec<i32>, k: i32) -> i32 {
    let k = k as usize;
    let mut base = tickets[k as usize];
    let mut ans: i32 = 0;
    (0..=k).for_each(|idx| {
      ans += base.min(tickets[idx]);
    });

    base -= 1;
    (k + 1..tickets.len()).for_each(|idx| {
      ans += base.min(tickets[idx]);
    });
    ans
  }
}
